src/full/Agda/Interaction/Library.hs

```diff
@@ -185,6 +203,17 @@ defaultLibraryFiles = ["libraries-" ++ version, "libraries"]
 defaultsFile :: FilePath
 defaultsFile = "defaults"
 
+-- | The @~/.agda/executables@ file lists the executables Agda should know about.
+--   The content of @executables@ is a list of paths to executables.
+--
+--   Agda honors also version specific @executables@ files, e.g. @executables-2.6.0@.
+--
+--   @defaultExecutablesFiles@ gives a list of all @executables@ Agda should process
+--   by default.
+--
+defaultExecutableFiles :: [FilePath]
```

Comment:
Use `Agda.Utils.List1`.

Reply:
While I understand where you're coming from, the use of list here directly follows the example set by the code in `Agda.Interaction.Library`. For instance:
```haskell
defaultLibraryFiles :: [FilePath]
defaultLibraryFiles = ["libraries-" ++ version, "libraries"]
```
I think consistency is probably more important, and while I would be in favor if fixing this more generally, I don't think that a new-feature pull request would be the place to fix unrelated issues.

Reply:
I'm happy to make these changes in a different pull request later on.

Reply:
Well, I am not asking to revamp the libraries framework at this point.
I'd say if you add now code, you would be free to use the latest insights even if the existing code doesn't.  The `List1` typing allows you to use the safe `List1.last` function rather than the a priori unsafe `List.last` one.  It saves the code reader from going back to check that the respective list is indeed non-empty.

Reply:
Would it be OK if I made the recommended changes (`List1`, `List1.last`, and a minor comment saying the list is small) to both my code and the existing code? I'd feel uncomfortable making the changes only to my code, and creating inconsistencies.

Reply:
Sure, would be very welcome!

Reply:
I've made the changes you suggested, for both my code and for the equivalent libraries code.